test/helper/launcher/wait.go

```diff
@@ -0,0 +1,28 @@
+package launcher
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	NoWait = (*WaitConfig)(nil)
+)
+
+type WaitConfig struct {
+	Condition string
+	Target    string
+	Timeout   time.Duration
+}
+
+func WaitReady(target string, timeout time.Duration) *WaitConfig {
+	return &WaitConfig{Condition: "condition=Ready", Target: target, Timeout: timeout}
+}
+
+func (cfg *WaitConfig) waitArgs() []string {
+	args := []string{"wait", fmt.Sprintf("--for=%s", cfg.Condition), cfg.Target}
+	if cfg.Timeout != 0 {
+		args = append(args, fmt.Sprintf("--timeout=%v", cfg.Timeout))
+	}
+	return args
```

Comment:
hmmmmm, not sure we should go into the business of building an abstraction framework over `kubectl` invocations. This is not the focus of the challenge at hand.

Not a blocker but I am sceptical.

Reply:
Again, this is not part of the public API of the launcher. We are in the business of taking a YAML and making it real in Atlas. This is one way of doing it, there can be other better ones we can figure out later. This unblocks us for now to progress, but can be replaced later as needed.

Reply:
> this is not part of the public API

ok, as long as this stays private, I am fine.